### PR TITLE
CA-1003 separating participants parameters (role, fullName etc..)

### PR DESCRIPTION
**Pull request by Galzk**

### CODE REVIEW

File: src/components/AudioLevelIndicator/AudioLevelIndicator.tsx [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
 import { interval } from 'd3-timer';
 import useIsTrackEnabled from '../../hooks/useIsTrackEnabled/useIsTrackEnabled';
 import useMediaStreamTrack from '../../hooks/useMediaStreamTrack/useMediaStreamTrack';
 import { PLAYER_STATE } from '../../utils/displayStrings';
 import { IconButton } from '@material-ui/core';
-import { ParticipantIdentity } from '../../utils/participantIdentity';
 import { getParticipantOptions } from '../../components/ParticipantInfo/ParticipantDropDown/ParticipantDropDown';
 import { observer } from 'mobx-react-lite';
 import rootStore from '../../stores/rootStore';
+import { ParticipantExtended } from '../../types/participantInformation';
```
Comment (talr) [inline src/components/AudioLevelIndicator/AudioLevelIndicator.tsx]:
```suggestion
import { ParticipantExtended } from '../../types/participantExtended';
```

File: src/components/Controls/EndCallButton/EndCallButton.test.tsx [hunk 2]
```diff
@@ -32,7 +32,7 @@ describe('Hang Up button', () => {
   });
 
   it('should call leave conference when clicked as a non-moderator', () => {
-    let participant = new mockLocalParticipant('test', 'Parent', 2);
+    let participant = new mockLocalParticipant('test', 'Parent', '2');
```
Comment (talr) [inline src/components/Controls/EndCallButton/EndCallButton.test.tsx]: Please use meaningful values i.e. "Dan", "Davidi"
Please use strongly types (FirstNameType, LastNameType, PersonIdType, UserIdType)

**Review by talr — CHANGES_REQUESTED** [review]

File: src/components/ParticipantStrip/ParticipantGrid/ParticipantGrid.test.tsx [hunk 3]
```diff
@@ -33,25 +33,25 @@ describe('the ParticipantStrip component', () => {
...
 
   it('should correctly render ParticipantInfo components', () => {
-    const localParticipant = new mockLocalParticipant('local', 'Reporter', 123, '2');
+    const localParticipant = new mockLocalParticipant('local', 'Reporter', '123', '2');
     rootStore.participantsStore.localParticipant?.setParticipant(localParticipant);
     // @ts-expect-error
     rootStore.participantsStore.setAudioTrack(mockLocalTrack);
-    const participant1 = new mockParticipant('participant1', 'Reporter', 1, '1');
-    const participant2 = new mockParticipant('participant2', 'Reporter', 2, '2');
+    const participant1 = new mockParticipant('participant1', 'Reporter', '1', '1');
```
Comment (talr) [inline src/components/ParticipantStrip/ParticipantGrid/ParticipantGrid.test.tsx]: Let change all Ids (i.e. UserId, PersonId ....) to UserIdType, PersonIdType

File: src/stores/localParticipantStore.ts [hunk 4]
```diff
@@ -16,19 +17,18 @@ class LocalParticipantStore {
...
     this.participantsStore = participantsStore;
   }
-  setParticipant(participant?: LocalParticipant) {
-    this.participant = participant;
+  setParticipant(participant?: LocalParticipantExtended) {
+    if (!participant) {
+      this.participant = undefined;
+      return;
+    }
+    const extendedParticipant = participant?.role ? participant : extendLocalParticipant(participant);
```
Comment (talr) [inline src/stores/localParticipantStore.ts]: Do we need it?

File: src/types/participantInformation.ts [hunk 5]
```diff
@@ -1,7 +1,17 @@
...
 export interface ParticipantInformation {
   caseReference: string;
   displayName: string;
   role: string;
   userId: number | null;
   videoConferenceRoomName: string;
 }
+
+export interface ParticipantExtended extends Participant {
+  role: string;
```
Comment (talr) [inline src/types/participantInformation.ts]:
```suggestion
  legalCaseRole: legalCaseRoleType;
```

File: src/utils/participantIdentity.ts [hunk 6]
```diff
@@ -27,3 +31,24 @@ export class ParticipantIdentity {
...
+  participant.personId = parsed.personId;
+  participant.userId = parsed.userId;
+  participant.isRegistered = !!parsed.userId;
+  return participant as ParticipantExtended;
+};
+export const extendLocalParticipant = (p: LocalParticipant) => {
+  const parsed = ParticipantIdentity.Parse(p?.identity || '');
+  let participant = p as LocalParticipantExtended;
+  participant.role = parsed.role;
+  participant.fullName = parsed.partyName;
```
Comment (talr) [inline src/utils/participantIdentity.ts:50]: I suggest to split to first name and last name

**Review by talr — CHANGES_REQUESTED** [review]

File: src/types/participantInformation.ts (hunk 5 above)
Comment (Galzk) [reply src/types/participantInformation.ts]: Let's do that when we need it to happen, leaving it as is for now

File: src/utils/participantIdentity.ts (hunk 6 above)
Comment (Galzk) [reply src/utils/participantIdentity.ts:50]: We should discuss where to implement this,
I have some questions:
- Do we keep the fullName variable?
- Do we separate the fullName ourselves to first and last, or do we count on the backend to provide that?
- What do we do when we only have fullName from the backend?
- Should we add prefix ("Dr", "Mr" etc)? What about middle name?

File: src/stores/roomsStore.ts [hunk 7]
```diff
@@ -173,7 +174,7 @@ class roomsStore {
       };
     } catch (err) {
       console.log(err.message);
```
Comment (talr) [inline src/stores/roomsStore.ts]: TODO replace with eScribers Logger

**Review by talr — APPROVED** [review]

**Pull request merged**